File: js/pigLatin.js
exports.translate = function(words) {
  let ans = [];
  let vowels = "aeiou"
  for (let word of  words.split(" ")) {
    let consonants ="";
    let end ="";
    if (!(/[a-zA-Z]/).test(word[word.length-1])) {
      end += word[word.length-1];
      word = word.slice(0,word.length-1)
    }
    for (let char of word) {
      if (char == "u" && consonants[-1] == "q") {
        consonants += char.toLowerCase()
        break
      } else if (vowels.includes(char)) {
        break
      }else{
        consonants += char.toLowerCase()
      }
    }
    if (word[0].toUpperCase() == word[0]) {
      ans.push(word[consonants.length].toUpperCase() +word.slice(consonants.length) + consonants + "ay" + end)
    }else{
      ans.push(word.slice(consonants.length) + consonants + "ay" + end)
    }
  }
  return ans.join(" ")
};

File: python/pig_latin.py
def translate(word_or_phrase):
  if not isinstance(word_or_phrase,str):
    print("Not a string")
  vowels = "aeiou"
  ans = []
  for word in word_or_phrase.split(" "):
    word = list(word)
    consonants = []
    end = []
    if not word[-1].isalpha():
      # for punctuation at the end of the word
      end.append(word[-1])
      word = word[:-1]
    for char in word:
      if char == "u" and consonants[-1] == "q":
        consonants.append(char.lower())
        break
      elif char in vowels:
        break
      consonants.append(char.lower())
    if word[0].upper() == word[0]:
      ans.append(word[len(consonants)].upper() + "".join(word[len(consonants)+1:]) + "".join(consonants) + "ay" + "".join(end))
    else:
      ans.append("".join(word[len(consonants):]) + "".join(consonants) + "ay" + "".join(end))
  return " ".join(ans)